### DISCUSSION TITLE
What can be the purpose of OAuth on your website or project?

### DISCUSSION THREAD

**Discussion by larswoltersdev** [community/community#24363] [Apps, API and Webhooks]
<p>I was wondering, what would be the purpose of creating a OAuth application and authentication system for your website or project. Of course to log in with it, but then what? <strong>What would be examples of good purposes to use GitHub OAuth, for example when doing a website application project?</strong></p>

**Comment by lee-dohm** [comment, marked as answer]
<p>As it states on the <a href="https://oauth.net">OAuth website</a>:</p>
<blockquote>
<p>OAuth is a simple way to publish and interact with protected data.</p>
</blockquote>
<p>One way to use GitHub OAuth in a non-GitHub website or application is as an identity mechanism. You give your application’s users the ability to use GitHub to authenticate who they are. This saves you from having to deal with passwords and all the mess that goes along with them. It also gives you an avatar for them as well, so you don’t have to store and police potentially abusive avatar images. If you’ve seen a “Sign in with GitHub” button on a website such as <a href="https://travis-ci.org">https://travis-ci.org</a> or <a href="https://exercism.io">https://exercism.io</a>, then that’s what they’re using GitHub OAuth for. Additionally, we have <a href="https://docs.github.com/free-pro-team@latest/rest/guides/basics-of-authentication">information on how to do this</a> in our developer documentation.</p>
<p>But how to use OAuth depends on what your website or application does.  Does your website or application need to see its user’s GitHub data? Then GitHub OAuth might be the right solution. Does your website or application need to calculate spaceship transfer orbits between Venus and Mars? Then GitHub OAuth is less likely to be the right solution.</p>
<p>I hope that helps!</p>
